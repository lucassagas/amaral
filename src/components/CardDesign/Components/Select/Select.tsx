import { Box, Select as ChackraSelect } from '@chakra-ui/react'
import React from 'react'

interface ISelect {
  w: string
  value: number
  setValue: (data: number) => void
  options: {
    value: number
    label: string
  }[]
}

export const Select = ({ w, options, value, setValue }: ISelect) => {
  return (
    <Box pt="13px">
      <ChackraSelect
        w={w}
        value={value}
        fontSize="14px"
        fontWeight="bold"
        borderRadius="10px"
        focusBorderColor="#1A1A1A"
        borderColor="#A9A9A9"
        onChange={(e) => setValue(parseInt(e.target.value))}
      >
        {options.map((option) => {
          return (
            <option key={option.value} value={option.value}>
              {option.label}
            </option>
          )
        })}
      </ChackraSelect>
    </Box>
  )
}
