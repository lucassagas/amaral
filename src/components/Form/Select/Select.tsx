import {
  FormControl,
  FormLabel,
  Select as ChakraSelect,
  SelectProps as ChakraSelectProps,
  Text,
} from '@chakra-ui/react'
import { FieldValues, UseFormRegister } from 'react-hook-form'

interface SelectProps extends ChakraSelectProps {
  name: string
  label?: string
  required?: boolean
  errors?: { [x: string]: any }
  register: UseFormRegister<FieldValues>
  options: Array<{
    name: string
    value: string
  }>
}

export function Select({
  name,
  label,
  register,
  errors,
  options,
  required = false,
  ...rest
}: SelectProps) {
  return (
    <FormControl>
      {label && (
        <FormLabel fontSize="sm" htmlFor="email">
          {label}
        </FormLabel>
      )}
      <ChakraSelect
        id={name}
        focusBorderColor="orange.500"
        bgColor="gray.900"
        variant="filled"
        isInvalid={errors && errors[name]}
        _hover={{ bgColor: 'gray.900' }}
        size="md"
        {...register(name, { required })}
        {...rest}
      >
        {options?.map((option) => (
          <option key={option.name} value={option.value}>
            {option.name}
          </option>
        ))}
      </ChakraSelect>
      {errors && errors[name] && (
        <Text fontSize="sm" color="red.500">
          Campo obrigat√≥rio
        </Text>
      )}
    </FormControl>
  )
}
