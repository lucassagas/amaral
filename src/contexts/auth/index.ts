export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
