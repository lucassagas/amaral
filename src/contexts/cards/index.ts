export { CardContext, CardContextProvider } from './cards';

