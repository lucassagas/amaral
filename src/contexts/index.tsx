import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
