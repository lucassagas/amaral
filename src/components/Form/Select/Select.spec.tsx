import { AppProvider } from '@/contexts/index'
import { render, screen } from '@testing-library/react'
import { Select } from './Select'

const register = jest.fn()

const renderComponent = () => {
  return render(
    <AppProvider>
      <Select
        name="name"
        label="Nome"
        register={register}
        options={[{ name: 'John Doe', value: '1' }]}
      />
    </AppProvider>
  )
}

describe('<Select />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  it('should render component with error', () => {
    render(
      <AppProvider>
        <Select
          name="name"
          label="Nome"
          register={register}
          options={[{ name: 'John Doe', value: '1' }]}
          errors={{ name: 'required' }}
        />
      </AppProvider>
    )

    expect(screen.getByText(/campo obrigat√≥rio/i)).toMatchSnapshot()
  })
})
