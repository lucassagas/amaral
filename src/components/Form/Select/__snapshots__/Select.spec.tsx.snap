// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render component with error 1`] = `
<p
  class="chakra-text css-1o0kj4s"
>
  Campo obrigat√≥rio
</p>
`;

exports[`<Select /> should render correctly 1`] = `
<div>
  <div
    class="chakra-form-control css-1kxonj9"
    role="group"
  >
    <label
      class="chakra-form__label css-pk5tju"
      for="email"
      id="field-1-label"
    >
      Nome
    </label>
    <div
      class="chakra-select__wrapper css-42b2qy"
    >
      <select
        class="chakra-select css-s5zj6t"
        id="name"
      >
        <option
          value="1"
        >
          John Doe
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-zj2dqk"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
